core/src/desul/atomics/Compare_Exchange_OpenMP.hpp

```diff
@@ -110,7 +121,7 @@ std::enable_if_t<Impl::atomic_always_lock_free(sizeof(T)),T> atomic_compare_exch
 template <typename T, class MemoryOrder, class MemoryScope>
 std::enable_if_t<!Impl::atomic_always_lock_free(sizeof(T)) && (sizeof(T)==16),T> atomic_compare_exchange(
     T* dest, T compare, T value, MemoryOrder, MemoryScope) {
-  if constexpr (omp_is_initial_device()) {
+  if constexpr (omp_on_host()) {
```

Comment:
Why can't you use `omp_is_initial_device` anymore?

Reply:
its not constexpr guaranteed, and in a newer version of clang it is not.